plugins/docker/attacher.go

```diff
@@ -137,12 +216,36 @@ func (m *AttachManager) attach(id string) {
 	fields["ContainerID"] = id
 	fields["ContainerName"] = name
 
+	// NOTE! Anything that is a duplicate key will be overridden with the value
+	// that is in the label, not the environment
+	for _, key := range m.fieldsFromLabels {
+		if value, ok := container.Config.Labels[key]; ok {
+			fields[key] = value
+		}
+	}
+
+	return fields, nil
+}
+
+// Attach to the output of a single running container
+func (m *AttachManager) attach(id string, client DockerClient) error {
+	m.ir.LogMessage(fmt.Sprintf("Attaching container: %s", id))
+
+	fields, err := m.extractFields(id, client)
+	if err != nil {
+		m.ir.LogError(err)
+		return err
```

Comment:
There are a lot of places where an error is logged before being returned. With the `withRetry` decorator pattern, it's a bit hard to keep track of where all of the error logging happens, but I'm pretty sure that there are more duplicates than just the ones already mentioned. Generally I recommend pushing the logging to the topmost layer that touches the error.